tidb-cloud/vector-search-integrate-with-amazon-bedrock.md

```diff
@@ -0,0 +1,310 @@
+        "accept": "application/json",
+        "body": {
+            "prompt": prompt,
+            "temperature": 0
+        }
+    }
+
+    body_bytes = json.dumps(payload['body']).encode('utf-8')
+
+    response = bedrock_runtime.invoke_model(
+        body=body_bytes,
+        contentType=payload['contentType'],
+        accept=payload['accept'],
+        modelId=payload['modelId']
+    )
+
+    result_body = json.loads(response.get("body").read())
+    completion = result_body["generation"]
+    return completion
+```
+
+### Step 6. Create a vector table
+
+In `demo.py`, add the following code to create a vector table to store text and vector embeddings:
+
+```python
+# ---- TiDB Setup and Vector Index Creation ----
+class Entity(Base):
+    """Define the Entity table with a vector index."""
+    __tablename__ = "entity"
+    id = Column(Integer, primary_key=True)
+    content = Column(Text)
+    content_vec = Column(VectorType(dim=dim_of_embedding_model), comment="hnsw(distance=l2)")
+
+# Create the table in TiDB
+Base.metadata.create_all(engine)
+```
+
+### Step 7. Save the vector data to TiDB Cloud Serverless
+
+In `demo.py`, add the following code to save the vector data to your TiDB Cloud Serverless cluster:
+
+```python
+# ---- Saving Vectors to TiDB ----
+def save_entities_with_embedding(session, contents):
+    """Save multiple entities with their embeddings to the TiDB Serverless database."""
+    for content in contents:
+        entity = Entity(content=content, content_vec=embedding(content))
+        session.add(entity)
+    session.commit()
+```
+
+### Step 8. Run the application
+
+1. In `demo.py`, add the following code to establish a database session, save embeddings to TiDB, ask an example question (such as "What is TiDB?"), and generate results from the model:
+
+    ```python
+    if __name__ == "__main__":
+        # Establish a database session
+        with Session(engine) as session:
+            # Example data
+            contents = [
+                "TiDB is a distributed SQL database compatible with MySQL.",
+                "TiDB supports Hybrid Transactional and Analytical Processing (HTAP).",
+                "TiDB can scale horizontally and provides high availability.",
+                "Amazon Bedrock allows seamless integration with foundation models.",
+                "Meta Llama 3 is a powerful model for text generation."
+            ]
+
+            # Save embeddings to TiDB
+            save_entities_with_embedding(session, contents)
+
+            # Example query
+            query = "What is TiDB?"
+            info_str = " ".join(contents)
+
+            # Generate result from Meta Llama 3
+            result = generate_result(query, info_str)
+            print(f"Generated answer: {result}")
+    ```
+
+2. Save all changes to `demo.py` and run the script:
+
+    ```shell
+    python3 demo.py
+    ```
+
+    The expected output is similar to the following:
+
+    ```
+    python3 main.py
```

Comment:
```suggestion
    python3 demo.py
```

Reply:
The "python3 main.py" in the https://www.pingcap.com/blog/building-generative-ai-applications-with-tidb-and-amazon-bedrock blog seems incorrect